chainer/functions/evaluation/r2_score.py

```diff
@@ -0,0 +1,62 @@
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class R2_score(function.Function):
+    def __init__(self, sample_weight, multioutput):
+        if sample_weight is not None:
+            raise NotImplementedError()
+        if multioutput in ['uniform_average', 'raw_values']:
+            self.multioutput = multioutput
+        else:
+            raise ValueError("invalid multioutput argument")
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+        pred_type, true_type = in_types
+
+        type_check.expect(
+            pred_type.dtype.kind == 'f',
+            true_type.dtype.kind == 'f'
+        )
+
+        type_check.expect(
+            pred_type.ndim >= true_type.ndim,
```

Comment:
We do not need this condition because `pred_type.shape == true_type.shape` implies the equality of `ndim`'s